src/main/groovy/io/seqera/wave/service/blob/impl/KubeTransferStrategy.groovy

```diff
@@ -61,63 +61,59 @@ class KubeTransferStrategy implements TransferStrategy {
         // run the transfer job
-        final result = transfer0(info, command, jobName)
-        // delete job
-        cleanupJob(jobName, result.exitStatus)
-        return result
+        k8sService.transferJob(info.jobName, blobConfig.s5Image, command, blobConfig)
     }
 
-    protected BlobCacheInfo transfer0(BlobCacheInfo info, List<String> command, String jobName) {
-        final job = k8sService.transferJob(jobName, blobConfig.s5Image, command, blobConfig)
-        final timeout = Math.round(blobConfig.transferTimeout.toMillis() *1.1f)
-        final podList = k8sService.waitJob(job, timeout)
-        final size = podList?.items?.size() ?: 0
-        // verify the upload pod has been created
-        if( size < 1 ) {
-            log.error "== Blob cache transfer failed - unable to schedule upload job: $info"
-            return info.failed("Unable to scheduler transfer job")
+    @Override
+    void cleanup(BlobCacheInfo blob) {
+        if( cleanup.shouldCleanup(blob.exitStatus) ) {
+            CompletableFuture.supplyAsync (() -> k8sService.deleteJob(blob.jobName), executor)
         }
-        // Find the latest created pod among the pods associated with the job
-        final latestPod = K8sHelper.findLatestPod(podList)
-
-        final pod = k8sService.getPod(latestPod.metadata.name)
-        final exitCode = k8sService.waitPodCompletion(pod, timeout)
-        final stdout = k8sService.logsPod(pod)
-
-        return exitCode!=null
-                ? info.completed(exitCode, stdout)
-                : info.failed(stdout)
     }
 
-    protected void cleanupJob(String jobName, Integer exitCode) {
-        if( cleanup.shouldCleanup(exitCode) ) {
-            CompletableFuture.supplyAsync (() -> k8sService.deleteJob(jobName), executor)
+    @Override
+    Transfer status(BlobCacheInfo info) {
+        final status = k8sService.getJobStatus(info.jobName)
+        if( !status || !status.completed() ) {
+            return new Transfer(mapToStatus(status))
         }
-    }
 
-    protected static String getJobName(BlobCacheInfo info) {
-        return 'transfer-' + Hashing
-                .sipHash24()
-                .newHasher()
-                .putUnencodedChars(info.locationUri)
-                .putUnencodedChars(info.creationTime.toString())
-                .hash()
+        // Find the latest created pod among the pods associated with the job
+        final pod = k8sService.getLatestPodForJob(info.jobName)
+        if( !pod )
+            throw new IllegalStateException("Missing carried pod for job: ${info.jobName}")
+
+        // determine exit code and logs
+        final exitCode = pod
+                .status
+                ?.containerStatuses
+                ?.first()
+                ?.state
+                ?.terminated
+                ?.exitCode
+        final stdout = k8sService.logsPod(pod)
+        return new Transfer(mapToStatus(status), exitCode, stdout)
```

Comment:
Considering the pod run by the job is expected to be terminated, i've tried to simplify this part. @munishchouhan can you please review and test it?

Reply:
ok sure